### PR TITLE
Add DrawShadowPassTransparent

### PR DESCRIPTION
**Pull request by lhog**

### CODE REVIEW

**Comment by badosu** [comment]
Would be interesting to have an idea of what are the possible applications for this callin, better so if documented undef `@function`

**Comment by badosu** [comment]
Yes: https://beyond-all-reason.github.io/spring/ldoc/modules/LuaHandle.html#Draw__Functions

**Comment by badosu** [comment]
A lot of stuff on LDoc needs to be figured out to be more clear and sectioned out properly. I consider not being able to identify the callins section a failure of the way I did it.

File: rts/Lua/LuaHandle.cpp [hunk 1]
```diff
@@ -2392,6 +2392,12 @@ DRAW_CALLIN(DrawPreDecals)
  */
 DRAW_CALLIN(DrawWorldPreParticles)
 
+/*** This callin has z and color buffer of shadowmap bound via FBO as well as the FFP state to do "semi-transparent" shadows pass (traditionally only used to draw particles).
```
Comment (badosu) [inline rts/Lua/LuaHandle.cpp:2395]: Just a minor remark. Use the text above `@function` for the one-liner descriptor. Below for extended description, e.g.:

```cpp
/*** Invoked when xyz
 *
 * @function Whatever
 * 
 * Does blablabl et blablabla, when lorem ipsum. Beware of harberblabber.
 */
```

Comment (lhog) [reply rts/Lua/LuaHandle.cpp:2395]: Ah ok, I was copying the syntax from somewhere else. Thanks for the hint.

Comment (badosu) [reply rts/Lua/LuaHandle.cpp:2395]: Yeah, currently the syntax is a bit over the place as it's an extended effort. It's not a problem, but once I work on the generator the one-liners will be more important (they are already used in the horrible default layout).

**Pull request merged**